### PR TITLE
Added a basic AggregateRoot

### PR DESCRIPTION
**Pull request by mumby0168**

### CODE REVIEW

File: src/Microsoft.Azure.CosmosEventSourcing/Aggregates/AggregateRoot.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    /// A method which applies a specific <see cref="DomainEvent"/>
+    /// </summary>
+    /// <param name="domainEvent"></param>
+    /// <example>
+    /// Implementing your own aggregates apply methods.
+    /// <code language="c#">
+    /// <![CDATA[
+    /// public class MyAggregate : AggregateRoot
+    /// {
+    ///     protected override void Apply(IDomainEvent domainEvent)
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosEventSourcing/Aggregates/AggregateRoot.cs:72]:
```suggestion
    ///     protected override void Apply(DomainEvent domainEvent)
```

File: src/Microsoft.Azure.CosmosEventSourcing/Aggregates/IAggregateRoot.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosEventSourcing.Events;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Aggregates;
+
+/// <summary>
+///
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosEventSourcing/Aggregates/IAggregateRoot.cs:9]: More missing comments... For public APIs we should really try to have the triple slash comments serve as a the docs.

**Review by IEvangelist — DISMISSED** [review]

File: src/Microsoft.Azure.CosmosEventSourcing/Aggregates/AggregateRoot.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosEventSourcing.Events;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Aggregates;
+
+/// <summary>
+///
```
[the remark this answers is not in the document]
Comment (mumby0168) [reply src/Microsoft.Azure.CosmosEventSourcing/Aggregates/AggregateRoot.cs:9]: Oops my bad, must have missed these ones.

File: src/Microsoft.Azure.CosmosEventSourcing/Aggregates/IAggregateRoot.cs (hunk 2 above)
Comment (mumby0168) [reply src/Microsoft.Azure.CosmosEventSourcing/Aggregates/IAggregateRoot.cs:9]: I have updated these.

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**